### PR TITLE
Decompile pppMove and pppPoint

### PR DESCRIPTION
**Pull request by Xeeynamo**
Depends on #7. I will rebase this once #7 is merged.

### CODE REVIEW

File: symbol_addrs.txt [hunk 1]
```diff
@@ -1968,3 +1968,10 @@ sceIpuInit                   = 0x0026f0a8; // type:func
 // hacks for now
 D_0048BAB8 = 0x0048BAB8; // type:s32
 D_004885C0 = 0x004885C0; // type:s32
+
+
+//=============================
+// ppp lib
+//=============================
+ppvMng                       = 0x006105A0;
+ppvUserStopPartF             = 0x0061095C; // type:s32
```
Comment (Yotona) [inline symbol_addrs.txt:1977]: I feel like it these should be with the rest of libppp's addrs following line 21

File: include/pppfunc.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+typedef struct _pppPObject {
+    /* 0x00 */ struct _pppPObject * next;
+    /* 0x04 */ struct _pppPObject * ppobj;
+    /* 0x08 */ pppPDataVal * pdtval;
+    /* 0x0C */ s32 time;
+    /* 0x10 */ FMATRIX cmat;
+    /* 0x50 */ FMATRIX dmat;
+    /* 0x90 */ struct pppSVECTOR * lpvert;
+    /* 0x94 */ struct pppSVECTOR * lpnorm;
```
Comment (Yotona) [inline include/pppfunc.h:75]: pppSVECTOR hasn't been defined yet

File: symbol_addrs.txt (hunk 1 above)
Comment (ethteck) [reply symbol_addrs.txt:1977]: I think keeping it by offset is ideal personally

File: include/ppp.h [hunk 3]
```diff
@@ -2,18 +2,22 @@
...
-    /* 0x08 */ void* dektFunc;
-    /* 0x0C */ void* drawFunc;
-    /* 0x10 */ void* iniFunc;
-    /* 0x14 */ void* staFunc;
-    /* 0x18 */ void* endFunc;
-    /* 0x1C */ void* conFunc;
-    /* 0x20 */ void* con2Func;
-    /* 0x24 */ void* desFunc;
-} pppProg; // size = 0x28
+    /* 0x00 */ u8 unk[0xB0];
```
Comment (ethteck) [inline include/ppp.h:8]: I prefer unk_00 (And to have underscores in general after unk)

File: include/pppfunc.h [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+#include "common.h"
+
+typedef f32 FMATRIX[4][4]; // size = 0x40
+
+typedef struct {
+    s32 time;
+} pppCDT;
+
+typedef struct {
+    short x, y, z;
```
Comment (ethteck) [inline include/pppfunc.h:13]: s16

File: include/pppfunc.h [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+} pppSVECTOR; // size = 0x6
+
+typedef struct {
+    float x, y, z;
```
Comment (ethteck) [inline include/pppfunc.h:17]: f32

File: src/ppp/pppMove.c [hunk 6]
```diff
@@ -1,5 +1,41 @@
...
+        move->y = move->y + p->vec.y;
+        move->z = move->z + p->vec.z;
+    }
+    v->vec.x = v->vec.x + move->x;
+    v->vec.y = v->vec.y + move->y;
+    v->vec.z = v->vec.z + move->z;
+}
+
+void pppMoveCon(pppPObject * pobj, pppCtrlTable * ctbl) {
+    float val = 0.0f;
```
Comment (ethteck) [inline src/ppp/pppMove.c:37]: f32

**Review by ethteck — CHANGES_REQUESTED** [review]
Left some naming comments - also I prefer left-aligned pointers...I really should set up the CI and format script. I'll try to do it soon, but I've never created a format script so it might take some time for me to figure out the right way to handle it

**Pull request merged**